pkg/index/index.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package index
+
+import (
+	"container/list"
+	"errors"
+	"fmt"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+var (
+	ErrNodeNotFound = errors.New("Node not found")
+)
+
+type Node interface {
+	// Returns the string representation (usually the ID) of the node
+	String() string
+
+	// Returns the string respresentation of the nodes parent (usually it's ID)
+	Parent() string
+
+	// Deep copy of the node
+	Copy() Node
+}
+
+type node struct {
+	Node
+	parent   *node
```

Comment:
Thats a good point.  Will rename to something better.  `Noder` sounds too much like neuter.  Maybe `Elementer` since the package and struct is called `Index`.